go/types/list.go

```diff
@@ -98,7 +116,10 @@ func (l List) Get(idx uint64) Value {
 
 type MapFunc func(v Value, index uint64) interface{}
 
+// Deprecated: This API may change in the future. Use IterAll or Iterator instead.
 func (l List) Map(mf MapFunc) []interface{} {
+	// TODO: This is bad API. It should have returned another List.
```

Comment:
It turns out https://github.com/attic-labs/noms/issues/2559 might be a better issue for this